servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsServiceDiscovererBuilder.java

```diff
@@ -68,9 +69,22 @@ public final class DefaultDnsServiceDiscovererBuilder implements DnsServiceDisco
     private static final int DEFAULT_MIN_TTL_CACHE_SECONDS = 0;
     private static final int DEFAULT_MAX_TTL_CACHE_SECONDS = 30;
     private static final int DEFAULT_TTL_POLL_JITTER_SECONDS = 4;
+    /**
+     * This is one of the standard <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/net/properties.html">
+     * Java Network Properties</a>.
+     * We do not support {@code networkaddress.cache.ttl} because it's behavior in JDK is different. It overrides the
+     * original TTL from the server and caches exactly for the specified amount of time instead of just being a max cap.
+     */
+    private static final String NEGATIVE_TTL_CACHE_SECONDS_PROPERTY = "networkaddress.cache.negative.ttl";
+    private static final int DEFAULT_NEGATIVE_TTL_CACHE_SECONDS;
     private static final ServiceDiscovererEvent.Status DEFAULT_MISSING_RECOREDS_STATUS = EXPIRED;
 
     static {
+        final Integer negativeCacheTtlValue = parseProperty(NEGATIVE_TTL_CACHE_SECONDS_PROPERTY);
+        DEFAULT_NEGATIVE_TTL_CACHE_SECONDS = negativeCacheTtlValue == null ? 0 :
+                // A value of -1 indicates "cache forever".
```

Comment:
The way it's coded it's a value less than 0 indicates "cache forever".

Reply:
Updated the comment to clarify